miio/waterheater.py

```diff
@@ -0,0 +1,219 @@
+    )
+    def set_target_temperature(self, temperature: int):
+        """Set target temperature."""
+        min_temp: int
+        max_temp: int
+        min_temp, max_temp = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["temperature_range"]
+        if not min_temp <= temperature <= max_temp:
+            raise WaterHeaterException("Invalid target temperature: %s" % temperature)
+        return self.send("set_temp", [temperature])
+
+    @command(default_output=format_output("Setting bacteriostatic mode on"))
+    def set_bacteriostatic_mode(self):
+        """Set bacteriostatic operational mode."""
+        mode: OperationMode
+        bacteriostatic_mode: bool
+        bacteriostatic_temp: int
+        bacteriostatic_mode = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["bacteriostatic_mode"]
+        if not bacteriostatic_mode:
+            raise WaterHeaterException("Bacteriostatic mode is not supported.")
+        bacteriostatic_temp = SUPPORTED_MODELS.get(
+            self.model, SUPPORTED_MODELS[MODEL_WATERHEATER_E1]
+        )["bacteriostatic_temperature"]
+        mode = OperationMode(self.send("get_prop", ["modeType"])[0])
+        if mode == OperationMode.Termostatic:
+            raise WaterHeaterException("Bacteriostatic operational mode is not supported in Termostatic mode.")
+        return self.send("set_temp", [bacteriostatic_temp])
+
+    @command(
+        click.argument("appoint_start", type=int),
+        click.argument("appoint_end", type=int),
+        default_output=format_output(
+             lambda appoint_end, appoint_start:
+            "Setting up the Booking mode operational interval from: %s " % appoint_start +
+            "to: %s " % appoint_end +
+            "(duration: %s)" % (appoint_end-appoint_start if appoint_end-appoint_start > 0 else appoint_end-appoint_start+24)
+        ),
+    )
+    def set_appoint(self, appoint_start, appoint_end):
+        """Setting up the Booking mode operational interval."""
+        if  0 <= appoint_start < 24 and 0 <= appoint_end < 24:
+            return self.send("set_appoint", [1, appoint_start, appoint_end])
+        else:
+            raise WaterHeaterException("Booking mode operational interval parameters must be within [0, 23].")
+
+    @command(
+        click.argument("mode", type=EnumType(OperationMode)),
+        default_output=format_output("Setting operation mode to {mode.name} ({mode.value})"),
+    )
+    def set_mode(self, mode: OperationMode):
+        """Set operation mode."""
+        appoint_start: int
+        appoint_end: int
+        if mode == OperationMode.Booking:
+            appoint_start = self.send("get_prop", ["appointStart"])[0]
+            appoint_end = self.send("get_prop", ["appointEnd"])[0]
+#            raise WaterHeaterException("Booking operational mode is not supported in set_mode use set_appoint instead of.")
```

Comment:
Should this be removed?

Reply:
Yes! Just a stub when the feature was not implemented.